src/Lexer/lexer.rs

```
@@ -615,7 +619,120 @@ impl Lexer {
+            }
+        }
+
+        // validate date (make sure month <= 12, day <= 31, per month/year basis)
+        let day: u16 = first_num_lexeme.parse().unwrap_or(0);
+        let month: u16 = format!("{}{}", digit1, digit2).parse().unwrap_or(0);
+        let year: u16 = format!("{}{}{}{}", digit3, digit4, digit5, digit6).parse().unwrap_or(0);
+
+        if let Err(reason) = Value::validate_date(day, month, year) {
+            return Err(CPSError {
+                error_type: ErrorType::Lexical,
+                message: format!("'{}' is not a valid date: {}", 
```

Clippy (1.97.1)

[warning] `format!` in `format!` args

(warning)